File: src/Pieces/Piece.java
package Pieces;
import Board.Board;

public class Piece {
    private boolean side;
    private int value;
    private String representation;

    public boolean getSide() {
        return side;
    }

    public int getValue() {
        return value;
    }

    public String getRepresentation() {
        return representation;
    }

    public void setRepresentation(String representation) {
        this.representation = representation;
    }

    public Piece(String representation) {
        this.representation = representation;
    }

    public Piece(boolean side, String representation) {
        this.side = side;
        this.representation = representation;
    }

    public Piece(boolean side, int value, String representation) {
        this.side = side;
        this.value = value;
        this.representation = representation;
    }

    public boolean checkMove(Board board, int [] move, boolean side) {
        return false;
    }
}

File: Python/src/Pieces/Piece.py
class Piece(): 
    def __init__(self, side, value, representation):
        self.side = side
        self.value = value
        self.representation = representation

class Empty(): 
    def __init__(self, representation):
        self.representation = representation
    
    def __str__(self): 
        return self.representation

    def checkMove(self, board, move, side):
        return False